packages/api/src/payments/stripePaymentProvider.ts

```diff
@@ -150,18 +158,43 @@ export class StripePaymentProvider extends BasePaymentProvider {
     }
   }
 
-  async checkIntentStatus(props: CheckIntentProps): Promise<IntentState> {
-    // TODO: fix this
-    /* const intent = await this.stripe.paymentIntents.retrieve(props.payment.intentID)
+  async checkIntentStatus({intentID}: CheckIntentProps): Promise<IntentState> {
+    const intent = await this.stripe.paymentIntents.retrieve(intentID)
+    const state = mapStripeEventToPaymentStatue(intent.status)
+
+    if (!state) {
+      logger('stripePaymentProvider').error(
+        'Stripe intent with ID: %s for paymentProvider %s returned with an unknown state %s',
+        intent.id,
+        this.id,
+        intent.status
+      )
+      throw new Error('unknown intent state')
+    }
+
+    if (!intent.metadata.paymentID) {
+      logger('stripePaymentProvider').error(
+        'Stripe intent with ID: %s for paymentProvider %s returned with empty paymentID',
+        intent.id,
+        this.id
+      )
+      throw new Error('empty paymentID')
+    }
+
+    let customerID
+    if (
+      intent.setup_future_usage === 'off_session' &&
+      intent.customer === null &&
+      intent.payment_method !== null
+    ) {
+      customerID = await this.createStripeCustomer({intent})
```

Comment:
Would be great to talk it over. I understand you `create customer` within a `check` method?

Reply:
Agree. Something we can discuss for refactoring.